build.gradle

```diff
@@ -416,3 +400,12 @@ gitPublish {
 gitPublishCopy {
     dependsOn 'updateGhPages', 'copyJavadoc', 'copyPicardDoc'
 }
+
+// For Gradle 8 explicitly add 'currentJar' as a dependency of the following tasks.
+// For more information, please refer to
+// https://docs.gradle.org/8.2.1/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.
+['distZip', 'distTar', 'startScripts', 'startShadowScripts', 'generateMetadataFileForPicardPublication'].each {
+    tasks.named(it).configure {
+        dependsOn 'currentJar'
+    }
+}
```

Comment:
Option b) Use a loop to put them all in one place

Reply:
It feels like there is probably a more idiomatic way to do this by adding it to the java component or something like that, but I don't know how and it doesn't seem worthwhile to figure it.   This is way better than a zillion individual closures adding it.